src/System.Net.Http/src/netcore50/System/Net/HttpClientHandler.cs

```diff
@@ -553,8 +555,15 @@ private void ApplyDecompressionSetting(HttpRequestMessage request, Decompression
             try
             {
                 await ConfigureRequest(request).ConfigureAwait(false);
-            
-                response = await this.handlerToFilter.SendAsync(request, cancellationToken).ConfigureAwait(false);
+
+                if (DiagnosticsHandler.IsEnabled())
+                {
+                    response = await this.diagnosticsPipeline.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    response = await this.handlerToFilter.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                }
```

Comment:
It would be more efficient in the resulting IL to do this as a single await rather than two:
```C#
Task<HttpResponseMessage> responseTask = DiagnosticsHandler.IsEnabled() ?
    this.diagnosticsPipeline.SendAsync(...) :
    this.handlerToFilter.SendAsync(...);
response = await responseTask.ConfigureAwait(false);
```